firstvoices/backend/tasks/import_job_tasks.py

```diff
@@ -215,6 +218,72 @@ def import_audio_resource(import_job, audio_data, dictionary_entry_data, dry_run
     return audio_import_result, dictionary_entry_data
 
 
+def import_img_resource(import_job, img_data, dictionary_entry_data, dry_run):
+    """
+    Imports image files and appends IDs of the imported files as related_images in dictionary_entry_data.
+    Returns updated dictionary_entry_data and result from image import.
+    """
+
+    img_import_result = ImageResource(
+        site=import_job.site,
+        run_as_user=import_job.run_as_user,
+        import_job=import_job.id,
+    ).import_data(dataset=img_data, dry_run=dry_run)
+
+    # Adding image ids
+    if img_import_result.totals["new"]:
+        img_lookup = {
+            row["img_filename"]: row["id"]
+            for row in img_data.dict
+            if row.get("img_filename")
+        }
+        dictionary_entry_data.append_col(
+            [""] * len(dictionary_entry_data), header="related_images"
+        )
+        related_img_col_index = dictionary_entry_data.headers.index("related_images")
+        for i, row in enumerate(dictionary_entry_data.dict):
+            img_filename = row.get("img_filename")
+            related_id = img_lookup.get(img_filename, "")
+            row_list = list(dictionary_entry_data[i])
+            row_list[related_img_col_index] = related_id  # comma separated string
+            dictionary_entry_data[i] = tuple(row_list)
+
+    return img_import_result, dictionary_entry_data
+
+
+def import_video_resource(import_job, video_data, dictionary_entry_data, dry_run):
+    """
+    Imports video files and appends IDs of the imported files as related_videos in dictionary_entry_data.
+    Returns updated dictionary_entry_data and result from video import.
+    """
+
+    video_import_result = VideoResource(
+        site=import_job.site,
+        run_as_user=import_job.run_as_user,
+        import_job=import_job.id,
+    ).import_data(dataset=video_data, dry_run=dry_run)
+
+    # Adding video ids
+    if video_import_result.totals["new"]:
+        video_lookup = {
+            row["video_filename"]: row["id"]
+            for row in video_data.dict
+            if row.get("video_filename")
+        }
+        dictionary_entry_data.append_col(
+            [""] * len(dictionary_entry_data), header="related_videos"
+        )
+        related_video_col_index = dictionary_entry_data.headers.index("related_videos")
+        for i, row in enumerate(dictionary_entry_data.dict):
+            video_filename = row.get("video_filename")
+            related_id = video_lookup.get(video_filename, "")
+            row_list = list(dictionary_entry_data[i])
+            row_list[related_video_col_index] = related_id  # comma separated string
+            dictionary_entry_data[i] = tuple(row_list)
+
+    return video_import_result, dictionary_entry_data
```

Comment:
Totally optional, but I know you were looking for ways to split up or organize this long file. The media import methods share a lot of logic but use different resources. They also depend on similar logic to "split" the csv data based on column names.

One option would be to move all that logic into a set of Importer classes that reuse the same methods and have internal data members to specify the column prefix, resource, etc. Then we could call the actual import methods similarly to:

`image_import_result, image_lookup = ImageImporter.import(import_job, csv, dry_run)`

Reply:
The could potentially also keep track of their own valid column header options, and maybe we could find a way to use those for the up-front validation of the csv file as well.